openpathsampling/sample.py

```diff
@@ -119,14 +121,15 @@ def __setitem__(self, key, value):
         self.append(value)
 
     def __eq__(self, other):
-        if len(self.samples) == len(other.samples):
-            for samp1 in self.samples:
-                if samp1 not in other.samples:
-                    return False
-
-            return True
-        else:
-            return False
+        return set(self.samples) == set(other.samples)
```

Comment:
`collections.Counter` is even better than a length check. My concern is that there are also `append` and `extend` methods for `SampleSet`, which could create a `SampleSet` where you'd get false equality from the set-based version. (Didn't there used to be an `apply` method which did the cleanup to make sure there was only one of each? There should be such a thing, but I don't see it in the code now.)